File: Questions/is_prime/is_prime.rs
use std::io;

struct DSA;

impl DSA {
    pub fn is_prime(a: i64) -> bool {
        let root = (a as f64).sqrt() as i64 + 1;
        for i in 2..root {
            if a % i == 0 {
                return false;
            }
        }
        true
    }
}

fn main() {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");
    let input: i64 = input.trim().parse().expect("Please type a number!");
    println!("Is prime number: {}", DSA::is_prime(input));
}

File: Topic-wise/Introduction/is_prime/is_prime.py
from math import isqrt
class DSA:
    def is_prime(self, n):
        root = isqrt(n) + 1
        for i in range(2, root):
            if n % i == 0:
                return False
        return True
        

dsa = DSA()
n = int(input())
res = dsa.is_prime(n)
print("Is prime number:", res)
